go.mod

```diff
@@ -152,3 +152,6 @@ require (
 	sigs.k8s.io/kustomize/kyaml v0.17.1 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.4.2 // indirect
 )
+
+// This is needed for update-codegen-clients.sh to pin the version of code-generator.
+replace k8s.io/code-generator => k8s.io/code-generator v0.31.0
```

Comment:
is this really effective?

Reply:
I.e. isn't the go install in the makefile enough?

Reply:
We can move this, but these would need to be "refactored" a bit:
https://github.com/kube-bind/kube-bind/blob/main/hack/update-codegen-clients.sh#L25-L33

```
CODEGEN_PKG=${CODEGEN_PKG:-$(cd "${SCRIPT_ROOT}"; go list -f '{{.Dir}}' -m k8s.io/code-generator)}

source "${CODEGEN_PKG}/kube_codegen.sh"
```

currently they are upstream style 🤷

Reply:
And this is root module, not APIS or clients. 
and once we move to 1.32 this will go away.

Reply:
oh, that are fake kinds? wtf